scalene/scalene_profiler.py

```diff
@@ -1189,20 +1187,11 @@ def should_trace(filename: str) -> bool:
         if "scalene/scalene" in filename:
             # Don't profile the profiler.
             return False
-        found_in_profile_only = False
-        for prof in profile_only_list:
-            if prof in filename:
-                found_in_profile_only = True
-                break
-
+        found_in_profile_only = any(prof in filename for prof in profile_only_list)
         if not found_in_profile_only:
             return False
         if Scalene.__args.profile_all:
             # Profile everything else, except for "only" choices.
-            found_in_profile_only = False
-            for prof in profile_only_list:
-                if prof in filename:
-                    return True
```

Comment:
Concerned about this logic here-- wasn't "profile-only" at one point used to exclude files, but that logic doesn't seem to be used anymore. What about that short circuit that used to be here?

Reply:
It is still there - it's just tested once now, instead of twice.

Reply:
By the time we get to this line, `found_in_profile_only` must be true.

Reply:
ok so I see that a particular optimization is unneeded because that loop has also been exorcised, but this still doesn't "profile everything else, except for "only" choices" (though it did not before)-- has this behavior been deprecated?

Reply:
Yeah, that is I think just a buggy comment!

Reply:
It should probably read "profile everything as long as it matches the profile-only strings, if any."

Reply:
It should probably read "profile everything as long as it matches the profile-only strings, if any."